include/deal.II/lac/trilinos_epetra_vector.h

```diff
@@ -0,0 +1,289 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__trilinos_epetra_vector_h
+#define dealii__trilinos_epetra_vector_h
+
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_TRILINOS
+
+#ifdef DEAL_II_WITH_MPI
+
+#include <deal.II/base/index_set.h>
+#include <deal.II/base/std_cxx11/shared_ptr.h>
+#include <deal.II/lac/read_write_vector.h>
+#include <deal.II/lac/vector_space_vector.h>
+#include "mpi.h"
+#include "Epetra_FEVector.h"
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  // Forward declaration
+  template <typename Number>
+  class ReadWriteVector;
+
+  namespace EpetraWrappers
+  {
+    /**
+     * This class implements a wrapper to the the Trilinos distributed vector
+     * class Epetra_FEVector. This class is derived from the
+     * LinearAlgebra::VectorSpaceVector class. Note however that Epetra only
+     * works with Number = double.
+     *
+     * @ingroup TrilinosWrappers
+     * @ingroup Vectors
+     * @author Bruno Turcksin, 2015
+     */
+    class Vector : public VectorSpaceVector<double>
+    {
+    public:
+      /**
+       * Constructor. Create a vector of dimension zero.
+       */
+      Vector();
+
+      /**
+       * Copy constructor. Sets the dimension and the partitioning to that of
+       * the given vector and copies all elements.
+       */
+      Vector(const Vector &V);
+
+      /**
+       * This constructor takes an IndexSet that defines how to distribute the
+       * individual components among the MPI processors. Since it also
+       * includes information abtou the size of the vector, this is all we
+       * need to genereate a %parallel vector.
+       */
+      Vector(const IndexSet &parallel_partitioner,
+             const MPI_Comm &communicator = MPI_COMM_WORLD);
```

Comment:
Yes, implied `MPI_COMM_WORLD` is not good. Let the user be explicit about the communicator to use. It's so incredibly hard to track down where something went wrong otherwise.